projects/bp-gallery/src/components/views/location-curating/tag-structure-helpers.tsx

```diff
@@ -192,3 +203,43 @@ export const useGetBreadthFirstOrder = (
 
   return tagOrder;
 };
+
+export const useGetChildMatrix = (flattenedTags: FlatTag[] | undefined) => {
+  const tagsById = useGetTagsById(flattenedTags);
+  const tagTree = useGetTagTree(flattenedTags, tagsById);
+  const topologicalOrder = useGetTopologicalOrder(tagsById);
+
+  const childMatrix = useMemo(() => {
+    if (!tagTree || !topologicalOrder || !flattenedTags) return;
+
+    const childrenMatrix = Object.fromEntries(
+      topologicalOrder.map(tag => [
+        tag.id,
+        Object.fromEntries(topologicalOrder.map(otherTag => [otherTag.id, false])),
+      ])
+    );
+
+    const visit = (tag: FlatTag) => {
```

Comment:
You could do `childrenMatrix[tag.id][tag.id] = true;` to avoid the parent-equals-child checks in `PictureMapView.tsx`